mbofui/app/js/controllers.js

```diff
@@ -143,23 +141,21 @@ ocellus.controller('mapController', ['$scope', '$rootScope','$filter', '$timeout
     }
   });
 
-  // handles category filtering
-  $('#categories a').on('click', function(e) {
-    // use the id of the link
-    // TODO: this needs to change - will need to have categories that will not validate as ids (spaces, etc.)
-    var category = $(this).attr('id');
-    if (category === 'all') {
+  $scope.filter_category = function(key){
+    if (key === 'all') {
       // set marker to unfiltered list
       $scope.markers = $scope.markersAll;
     } else {
       // reset to unfiltered list (voids previous filters)
       $scope.markers = $scope.markersAll;
       // use a filter to only show the selected category
       $scope.markers = $filter('filter')($scope.markers, {
```

Comment:
In place of this:

```javascript
      $scope.markers = $scope.markersAll;
      $scope.markers = $filter('filter')($scope.markers, {
        category: key
      });
```

The code could be a little more compact and still readable written as:

```javascript
      $scope.markers = $filter('filter')($scope.markersAll, {
        category: key
      });
```